### PR TITLE
Auxilliary changes from timeseries branch

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Port over useful changes from the timeseries branch. 

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as all of them.

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/5674/

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBKey.cpp [hunk 1]
```diff
@@ -310,9 +310,10 @@ TRI_voc_cid_t RocksDBKey::viewId(rocksdb::Slice const& slice) {
 }
 
 LocalDocumentId RocksDBKey::documentId(rocksdb::Slice const& slice) {
-  TRI_ASSERT(slice.size() == 2 * sizeof(uint64_t));
+  TRI_ASSERT(slice.size() >= 2 * sizeof(uint64_t));
   // last 8 bytes should be the LocalDocumentId
-  return LocalDocumentId(uint64FromPersistent(slice.data() + sizeof(uint64_t)));
+  const char* ptr = slice.data() + slice.size() - sizeof(uint64_t);
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBKey.cpp]: is this actually compatible with the previous code?

File: arangod/RocksDBEngine/RocksDBColumnFamily.h [hunk 2]
```diff
@@ -29,20 +29,18 @@
...
 /// counter manager. Maybe the the number of families in the shouldHandle method
-/// needs
-/// to be changed
+/// needs to be changed
 struct RocksDBColumnFamily {
   friend class RocksDBEngine;
 
   static constexpr size_t minNumberOfColumnFamilies = 7;
-  static constexpr size_t numberOfColumnFamilies = 7;
+  static constexpr size_t numberOfColumnFamilies = 8;
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBColumnFamily.h]: we do have a new column family? for what exactly?

File: arangod/RocksDBEngine/RocksDBTransactionState.cpp [hunk 3]
```diff
@@ -238,8 +239,24 @@ void RocksDBTransactionState::cleanupTransaction() noexcept {
 }
 
 arangodb::Result RocksDBTransactionState::internalCommit() {
-  TRI_ASSERT(_rocksTransaction != nullptr);
-
+  if (!hasOperations()) { // bail out early
+    TRI_ASSERT(_rocksTransaction == nullptr ||
+               _rocksTransaction->GetNumKeys() == 0 &&
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBTransactionState.cpp]: can we use parentheses here to make this instantly clear to the reader?

File: js/client/modules/@arangodb/arango-database.js [hunk 4]
```diff
@@ -347,7 +347,7 @@ ArangoDatabase.prototype._create = function (name, properties, type, options) {
       'doCompact', 'keyOptions', 'shardKeys', 'numberOfShards',
       'distributeShardsLike', 'indexBuckets', 'id', 'isSmart',
       'replicationFactor', 'minReplicationFactor', 'shardingStrategy', 'smartGraphAttribute',
-      'smartJoinAttribute', 'avoidServers', 'cacheEnabled'].forEach(function (p) {
+      'smartJoinAttribute', 'avoidServers', 'cacheEnabled', 'labels'].forEach(function (p) {
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-database.js]: do we want to include this in devel already?

File: js/client/modules/@arangodb/testsuites/arangosh.js [hunk 5]
```diff
@@ -102,7 +102,7 @@ function arangosh (options) {
     }
 
     const startTime = time();
-    let rc = executeExternalAndWait(pu.ARANGOSH_BIN, toArgv(args), false, 0 /*, coverageEnvironment() */);
+    let rc = executeExternalAndWait(pu.ARANGOSH_BIN, toArgv(args), false, 0, coverageEnvironment());
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testsuites/arangosh.js]: not sure if this is related

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/7337/

File: lib/Basics/StaticStrings.cpp [hunk 6]
```diff
@@ -45,6 +45,7 @@ std::string const StaticStrings::KeyString("_key");
 std::string const StaticStrings::RevString("_rev");
 std::string const StaticStrings::FromString("_from");
 std::string const StaticStrings::ToString("_to");
+std::string const StaticStrings::TimeString("_time");
```
Comment (danielhlarkin) [inline lib/Basics/StaticStrings.cpp]: Not necessary yet.

File: lib/Basics/StaticStrings.cpp [hunk 7]
```diff
@@ -106,6 +107,7 @@ std::string const StaticStrings::IndexNameEdgeFrom("edge_from");
 std::string const StaticStrings::IndexNameEdgeTo("edge_to");
 std::string const StaticStrings::IndexNameInaccessible("inaccessible");
 std::string const StaticStrings::IndexNamePrimary("primary");
+std::string const StaticStrings::IndexNameTime("time");
```
Comment (danielhlarkin) [inline lib/Basics/StaticStrings.cpp]: Not necessary yet.

File: lib/Basics/StaticStrings.h [hunk 8]
```diff
@@ -51,6 +51,7 @@ class StaticStrings {
   static std::string const RevString;
   static std::string const FromString;
   static std::string const ToString;
+  static std::string const TimeString;
```
Comment (danielhlarkin) [inline lib/Basics/StaticStrings.h]: Not necessary yet.

File: lib/Basics/StaticStrings.h [hunk 9]
```diff
@@ -105,6 +106,7 @@ class StaticStrings {
   static std::string const IndexNameEdgeTo;
   static std::string const IndexNameInaccessible;
   static std::string const IndexNamePrimary;
+  static std::string const IndexNameTime;
```
Comment (danielhlarkin) [inline lib/Basics/StaticStrings.h]: Not necessary yet.

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5694/

**Pull request merged**